test/e2e/node/ssh.go

```diff
@@ -49,6 +49,7 @@ var _ = SIGDescribe("SSH", func() {
 		if err != nil {
 			framework.Failf("Error getting node hostnames: %v", err)
 		}
+		e2eskipper.SkipUnlessAtLeast(len(hosts), 1, "Skipping test because found no SSH'able node")
```

Comment:
This fixes the SSH node test and skips the test if no SSH'able nodes are available.

Reply:
But by this point shouldn't we expect that there are SSH'able nodes and consider it an error if there are none?

/cc @wojtek-t @mm4tt
Would like input from sig-scalability as this is being point to as a fix for https://github.com/kubernetes/kubernetes/issues/103688

Reply:
Remove this line and I'm happy to approve if SIG Node is happy with it

Reply:
I would prefer to remove this line as well. I tried to understand reason behind failing this test in scalability job and not in regular tests. The difference is that in scalability jobs we don't have public IPs on our nodes and we use "gcp-ssh-proxy-instance-name" kubetes flag. This flag sets "KUBE_SSH_BASTION" env.  You can see here that there is special case for ssh when this env is set: https://github.com/kubernetes/kubernetes/blob/e3363152383ce8fc210290929e366ffdcbf71af9/test/e2e/framework/ssh/ssh.go#L210 
So I believe that we should take this into account when checking for list of available hosts here: https://github.com/kubernetes/kubernetes/blob/fac3dd6914f95e57975e7de7ccb3de2777f9edd6/test/e2e/framework/ssh/ssh.go#L143

Reply:
Agree - that's an incorrect change, we don't want this.
See this comment: https://github.com/kubernetes/kubernetes/pull/103640/files#r673923883